### PR TITLE
forwarding: add built-in tunnel forwarding extension

### LINKED ISSUE #189677: Support public/private port toggle in tunnel forwarding
@derekbekoe

> While testing [microsoft/vscode-internalbacklog#4382](https://github.com/microsoft/vscode-internalbacklog/issues/4382) via the instructions in [microsoft/vscode-remote-release#8751](https://github.com/microsoft/vscode-remote-release/issues/8751), I noticed that switching port visibility has no effect and the port stays "Private".
> 
> Public visibility would be helpful for scenarios such as testing webhooks.
> 
> _Also, there are a few vs code repos but hopefully this is an appropriate one to file this._
> 
> cc: @connor4312

### LINKED ISSUE #189678: Make tunnel forwarding built-in, instead of using separate remote-tunnels extension

### PR DESCRIPTION
**Pull request by connor4312**
- Support public/private ports, which accounts for most of the work in
  the CLI. Previously ports were only privat.
- Make the extension built-in. Ported from the remote-containers
  extension with some tweaks for privacy and durability.
- This also removes the opt-in flag, by not reimplementing it 😛

Fixes https://github.com/microsoft/vscode/issues/189677
Fixes https://github.com/microsoft/vscode/issues/189678

### CODE REVIEW

File: extensions/tunnel-forwarding/package.nls.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+	"displayName": "Local Tunnel Port Forwarding",
+	"description": "Allows forwarding local ports to be accesible over the internet.",
```
Comment (rzhao271) [inline extensions/tunnel-forwarding/package.nls.json:3]:
```suggestion
	"description": "Allows forwarding local ports to be accessible over the internet.",
```

File: extensions/tunnel-forwarding/src/extension.ts [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+		}
+	}
+
+	private updateActivePortsIfRunning() {
+		if (this.state.state !== State.Starting && this.state.state !== State.Active) {
+			return;
+		}
+
+		const ports = [...this.tunnels].map(t => ({ number: t.remoteAddress.port, privacy: t.privacy }));
+		this.state.process.stdin.write(`${JSON.stringify(ports)}\n`);
```
Comment (rzhao271) [inline extensions/tunnel-forwarding/src/extension.ts:208]: Not sure if this is to update the state considering it's stdin

**Review by rzhao271 — COMMENTED** [review]
LGTM after the comments are resolved

File: extensions/tunnel-forwarding/src/extension.ts (hunk 2 above)
Comment (connor4312) [reply extensions/tunnel-forwarding/src/extension.ts:208]: Yea, the forwarding process reads from its stdin to take updates to the forwarded ports

**Review by DonJayamanne — APPROVED** [review]
Code review comments have been actioned,

**Comment by rzhao271** [comment]
I'm putting up a PR to fix the Code OSS build

**Pull request merged**